Imports iniciales

In [8]:
import pandas as pd

Cargamos el dataset y hacemos una revision inicial de los datos para conocer sus dimensiones y sus columnas

In [9]:
#Carga el dataset
oferta_gastronomica_df = pd.read_csv("oferta_gastronomica_limpio.csv", sep=";", encoding="latin-1")

#Revision inicial datos
print("\nDIMENSION DEL DATASET")
print(f"Filas: {oferta_gastronomica_df.shape[0]}")
print(f"Columnas: {oferta_gastronomica_df.shape[1]}")
print("\nCOLUMNAS DEL DATASET")
print(oferta_gastronomica_df.columns.tolist())


DIMENSION DEL DATASET
Filas: 2823
Columnas: 18

COLUMNAS DEL DATASET
['long', 'lat', 'id', 'nombre', 'categoria', 'cocina', 'ambientacion', 'telefono', 'mail', 'horario', 'calle_nombre', 'calle_altura', 'calle_cruce', 'direccion_completa', 'barrio', 'comuna', 'codigo_postal', 'codigo_postal_argentino']


Exploramos los tipos de cocina y categorías para conocer la oferta. Seleccionamos aquellas correspondientes
a comida saludable y filtramos.

En la columna "Cocina" se encontró "NATURISTA" y "VEGETARIANA". Se optó por filtrar a partir de esos valores al ser los que más relación tienen con comida saludable.

In [10]:
## Explorar los tipos de cocina presentes en el dataset
print("\nTIPOS DE COCINA DISPONIBLES")
print(oferta_gastronomica_df["cocina"].value_counts())

## Explorar las categorías de establecimientos
print("\nCATEGORÍAS DISPONIBLES")
print(oferta_gastronomica_df["categoria"].value_counts())

#FILTRADO
rubros_saludables = [
    "NATURISTA",
    "VEGETARIANA"
]


TIPOS DE COCINA DISPONIBLES
cocina
MINUTAS                737
INTERNACIONAL          448
GASTRONOMIA GENERAL    359
TRADICIONAL            297
PIZZERIA               176
PARRILLA               159
SANDWICHERIA           127
FAST FOOD               81
ITALIANA                77
MEDITERRANEA            48
PORTEÑA                39
DE AUTOR                32
REGIONAL                30
ESPAÑOLA               25
CHINA                   17
SUSHI                   12
TENEDOR LIBRE           11
JAPONESA                10
AMERICANA                9
FUSION                   8
FRANCESA                 8
TAPAS                    8
NATURISTA                7
MEXICANA                 7
MARISCOS                 7
MODERNO                  7
CUBANA                   6
ARABE                    6
ARMENIA                  6
BOLIVIANA                6
GOURMET                  5
CLASICO PORTEÑO         4
ORIENTAL                 4
VEGETARIANA              4
PERUANA                  4
ALEMANA            

ANALISIS PRINCIPAL

Cantidad de establecimientos saludables.

Qué porcentaje representan.

Distribución por categoría/barrio.

Detección de barrios sin oferta saludable.

Locales saludables encontrados.

In [11]:
saludables_df = oferta_gastronomica_df[
    oferta_gastronomica_df["cocina"].isin(rubros_saludables)
].copy()

## Mostrar la cantidad de establecimientos saludables
cantidad_total = len(oferta_gastronomica_df)
cantidad_saludables = len(saludables_df)

porcentaje_saludables = (
    cantidad_saludables / cantidad_total * 100
)

print("\nCantidad de establecimientos saludables:")
print(cantidad_saludables)

print("\nPorcentaje sobre el total:")
print(round(porcentaje_saludables, 2), "%")

## Analizar la distribución por rubro
print("\nDISTRIBUCIÓN POR RUBRO")
print(saludables_df["cocina"].value_counts())

## Analizar la distribución por categoría
print("\nDISTRIBUCIÓN POR CATEGORÍA")
print(saludables_df["categoria"].value_counts())

## Analizar la concentración por barrio
print("\nCONCENTRACIÓN POR BARRIO")
print(saludables_df["barrio"].value_counts())

## Detectar barrios sin oferta saludable
todos_los_barrios = set(oferta_gastronomica_df["barrio"].unique())

barrios_saludables = set(saludables_df["barrio"].unique())

barrios_sin_oferta = sorted(
    todos_los_barrios - barrios_saludables
)

print("\nBARRIOS SIN OFERTA SALUDABLE")
for barrio in barrios_sin_oferta:
    print("-", barrio)

## Mostrar los locales saludables encontrados
print("\nLOCALES SALUDABLES ENCONTRADOS")

print(
    saludables_df[
        ["nombre", "categoria", "cocina", "barrio", "comuna"]
    ].to_string(index=False)
)


Cantidad de establecimientos saludables:
11

Porcentaje sobre el total:
0.39 %

DISTRIBUCIÓN POR RUBRO
cocina
NATURISTA      7
VEGETARIANA    4
Name: count, dtype: int64

DISTRIBUCIÓN POR CATEGORÍA
categoria
RESTAURANTE    10
CAFE            1
Name: count, dtype: int64

CONCENTRACIÓN POR BARRIO
barrio
Palermo        3
Recoleta       2
San Nicolas    2
Saavedra       1
Colegiales     1
Caballito      1
Floresta       1
Name: count, dtype: int64

BARRIOS SIN OFERTA SALUDABLE
- Agronomia
- Almagro
- Balvanera
- Barracas
- Belgrano
- Boca
- Boedo
- Chacarita
- Coghlan
- Constitucion
- Flores
- Liniers
- Mataderos
- Monserrat
- Monte Castro
- Nueva Pompeya
- Nuñez
- Parque Avellaneda
- Parque Chacabuco
- Parque Patricios
- Paternal
- Puerto Madero
- Retiro
- San Cristobal
- San Telmo
- Velez Sarsfield
- Versalles
- Villa Crespo
- Villa Del Parque
- Villa Devoto
- Villa Gral. Mitre
- Villa Lugano
- Villa Luro
- Villa Ortuzar
- Villa Pueyrredon
- Villa Real
- Villa Riachuelo
- Villa Santa

CONCLUSIONES FINALES

La oferta saludable representa aproximadamente un 0.39% del total analizado.

Los rubros "naturista" y "vegetariano" tienen una presencia reducida.

Los barrios sin establecimientos saludables registrados podrían presentar oportunidades para abrir nuevos locales.
También existe una oportunidad de ofrecer formatos distintos a los que actualmente predominan, combinando rubros poco representados, como lo es la cocina saludable en general, y categorías poco representadas, es decir, el tipo de local o servicio ofrecido. Por ejemplo:

- Delivery de comida saludable.
- Cafetería saludable.
- Comida rápida saludable.
- Comida por peso saludable.
- Sandwichería saludable.
